code/datums/gamemode/dynamic/dynamic_rulesets_roundstart.dm

```diff
@@ -286,15 +287,47 @@
 	var/datum/faction/malf/unction = find_active_faction_by_type(/datum/faction/malf)
 	if (!unction)
 		unction = ticker.mode.CreateFaction(/datum/faction/malf, null, 1)
-	var/mob/M = pick(candidates)
-	assigned += M
-	candidates -= M
+
+	var/mob/M = progressive_job_search()
+	if(M.mind.assigned_role != "AI")
+		for(var/mob/new_player/player in mode.candidates) //mode.candidates is everyone readied up, not to be confused with candidates
+			if(M.mind.assigned_role == "AI")
```

Comment:
shouldn't this be player.mind.assigned_role == "AI"

Reply:
yes good catch